ngraph/core/src/descriptor/tensor.cpp

```diff
@@ -82,6 +72,15 @@ const Shape& descriptor::Tensor::get_shape() const
 {
     if (m_partial_shape.is_static())
     {
+        if (m_shape_changed.load(std::memory_order_relaxed))
+        {
+            std::lock_guard<std::mutex> guard(shape_mutex);
+            if (m_shape_changed) // double check after mutex lock
```

Comment:
Have you check if https://en.cppreference.com/w/cpp/atomic/atomic/exchange will get speed up here?

Reply:
This block of code is considered to be executed not often and overall performance speed up will be very tiny here.
Also 'exchange' is probably not suitable here but some speedup could be achieved using `load(memory_order_acquire)` and `store(memory_order_release)`. But it will make code less readable and this will be not much faster as we're using mutexes and construction of 'Shape' in this block